icepyx/core/granules.py

```diff
@@ -261,13 +273,13 @@ def get_avail(self, CMRparams, reqparams, cloud=False):
     # DevGoal: add kwargs to allow subsetting and more control over request options.
     def place_order(
         self,
-        CMRparams,
-        reqparams,
+        CMRparams: CMRParams,
+        reqparams: EGISpecificParamsSearch,
```

Comment:
```suggestion
        reqparams: EGISpecificParamsDownload,
```

Reply:
is `Query.reqparams` always of type `EGISpecificParamsDownload`? Or only sometimes (this is my understanding)?

Reply:
bb96a5f

I think this is probably the wrong way to handle things. We're telling the typechecker "Ignore your eyes and ears, just trust us" with the `cast()` call. Instead, we can add a "typeguard" which is a runtime check that asserts a narrower type, e.g. a literal `assert isinstance(...)` or `if isinstance(...): raise`

Reply:
> is `Query.reqparams` always of type `EGISpecificParamsDownload`? Or only sometimes (this is my understanding)?

The latter - it's only sometimes of type `EGISpecificParamsDownload` (when `Query._reqparams._reqtype = "download"). Otherwise it's of type "search", which maps to our `EGIRequiredParamsSearch`.

> but perhaps we do know for sure that reqparams is always EGISpecificParamsDownload at the point we're cast()ing.

I think this is true.